src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Helpers.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Text;
+using System.Diagnostics;
+using System.Globalization;
+using System.Security.Cryptography;
+
+namespace Internal.Cryptography
+{
+    internal static class Helpers
+    {
+        public static byte[] CloneByteArray(this byte[] src)
+        {
+            return (byte[])(src.Clone());
+        }
+
+        // Encode a byte array as an upper case hex string.
+        public static String ToHexStringUpper(this byte[] bytes)
+        {
+            StringBuilder sb = new StringBuilder(bytes.Length * 2);
+            foreach (byte b in bytes)
+            {
+                sb.AppendFormat("{0:X2}", b);
+            }
+            return sb.ToString();
+        }
+
+        // Decode a hex string-encoded byte array passed to various X509 crypto api. The parsing rules are overly forgiving but for compat reasons,
+        // they cannot be tightened.
+        public static byte[] DecodeHexString(this String s)
+        {
+            String hexString = DiscardWhiteSpaces(s);
+            uint cbHex = (uint)hexString.Length / 2;
+            byte[] hex = new byte[cbHex];
+            int i = 0;
+            for (int index = 0; index < cbHex; index++)
+            {
+                hex[index] = (byte)((HexToByte(hexString[i]) << 4) | HexToByte(hexString[i + 1]));
+                i += 2;
+            }
+            return hex;
+        }
+
+        private static String DiscardWhiteSpaces(String inputBuffer)
+        {
+            return DiscardWhiteSpaces(inputBuffer, 0, inputBuffer.Length);
+        }
+
+        private static String DiscardWhiteSpaces(String inputBuffer, int inputOffset, int inputCount)
+        {
+            int i, iCount = 0;
+            for (i = 0; i < inputCount; i++)
+            {
+                if (Char.IsWhiteSpace(inputBuffer[inputOffset + i]))
+                {
+                    iCount++;
+                }
+            }
+
+            char[] rgbOut = new char[inputCount - iCount];
```

Comment:
Although it looks like the only place DiscardWhiteSpaces is used is in DecodeHexString, so if we could get rid of that dependency as noted earlier, this won't matter :smile: